java/src/main/java/org/whispersystems/signalservice/api/SignalServiceMessageSender.java

```diff
@@ -170,18 +170,30 @@ public void sendMessage(SignalServiceAddress recipient, SignalServiceDataMessage
   public void sendMessage(List<SignalServiceAddress> recipients, SignalServiceDataMessage message)
       throws IOException, EncapsulatedExceptions
   {
-    byte[]              content   = createMessageContent(message);
-    long                timestamp = message.getTimestamp();
-    SendMessageResponse response  = sendMessage(recipients, timestamp, content, true);
+    byte[]                  content                 = createMessageContent(message);
+    long                    timestamp               = message.getTimestamp();
+    EncapsulatedExceptions  encapsulatedExceptions  = null;
+
+    try {
+      sendMessage(recipients, timestamp, content, true);
+    } catch (EncapsulatedExceptions e) {
+      encapsulatedExceptions = e;
+    }
 
     try {
-      if (response != null && response.getNeedsSync()) {
+      if (!store.getSubDeviceSessions(localAddress.getNumber()).isEmpty()) {
```

Comment:
The server knows only how many devices i have and if the message is a selfmessage. The client knows that too, so the server's response is irrelevant.

In case of group messages, the response is always be true if you have linked devices, so we can just check if we have open subdevice sessions.